### ISSUE TITLE
[CP Request] Fix for using Impeller with multiple blurs

### ISSUE THREAD

**Issue by stx** [flutter/flutter#138865]
Our app relies heavily on multiple blur effects and is unusable with Impeller due to this issue, forcing us to ship with Skia. It is fixed on master using https://github.com/flutter/engine/pull/47808.

### Issue Link

https://github.com/flutter/flutter/issues/131567
https://github.com/flutter/flutter/issues/132735

### Commit Hash

a6c939f

### Target

stable

### PR Link

https://github.com/flutter/engine/pull/47808

### Changelog Description

[Impeller] Improved performance when multiple blurs are present

### Impacted Users

All Impeller users using blur effects.

### Impact Description

Bad jank compared to Skia when multiple blurs are present on screen.

### Workaround

Using Skia.

**Comment by CaseyHillers** [comment]
@jonahwilliams can you review this CP request? If it looks good to you, add the "cp: approved" label.

**Comment by CaseyHillers** [comment]
@stx before we can go ahead with this, we'll need a PR opened to the stable branch. Can you update your PR description with a PR pointing to flutter-3.16-candidate.0? Thanks!

**Comment by jonahwilliams** [comment]
Unfortunately the change requires a follow up change (https://github.com/flutter/engine/pull/48190) to avoid introducing regressions, which I think makes this risker. I can ask at the weekly what the temperature is on this change.

**Comment by CaseyHillers** [comment]
@jonahwilliams do you have any updates you can share here?

**Comment by jonahwilliams** [comment]
We've decided not to cherry pick this.

**Comment by stx** [comment]
@jonahwilliams Our app now has frequent Skia crashes that have been ongoing for over a year since Skia crashes have essentially been de-prioritized and ignored in favor of Impeller improvements (totally understandable) or _still_ unusable Impeller performance. So we've been dealing with a bunch of 1-star crash reviews or 1-star poor performance reviews for more than a year. This is on an existing large commercial app that previously had no prior issues on Flutter 2.x, but we were forced to upgrade to broken Skia or a broken Impeller in order to maintain compatibility with the latest iOS and Android releases. This feels pretty godawful. Just FYI for whoever cares about this over there.

**Comment by jonahwilliams** [comment]
> Our app now has frequent Skia crashes that have been ongoing for over a year since Skia crashes have essentially been de-prioritized

Sorry this just isn't true. What crashes/where have they been filled?

**Comment by jonahwilliams** [comment]
I don't really understand, what does this have to do with a blur performance issue?

**Comment by stx** [comment]
> I don't really understand, what does this have to do with a blur performance issue?

It's a response to your question about where our Skia crashes were...?

**Comment by jonahwilliams** [comment]
okay I see. So you want to update to Impeller, but can't because of this performance issue - and you're having trouble on the Skia backend due to this crashes?

**Comment by stx** [comment]
> okay I see. So you want to update to Impeller, but can't because of this performance issue - and you're having trouble on the Skia backend due to this crashes?

Yes, exactly. So existing apps are left choosing between crashes or low FPS.

I demarcate existing apps because new apps can design differently to avoid crashes and FPS drops. Larger existing apps can't just redo their design.

**Comment by jonahwilliams** [comment]
So you've tried this change out, and it has substantially improved performance for you? Both of the other folks I talked to on the original issue did not notice substantial differences.

**Comment by stx** [comment]
> So you've tried this change out, and it has substantially improved performance for you? Both of the other folks I talked to on the original issue did not notice substantial differences.

35ca279ce97605eeddc8d6179f2d6d8b1edd9b9c

https://github.com/flutter/flutter/assets/1284134/68d83b97-15f8-4366-85ba-7a95ddb3eb3c

8b2e5b30a7a1b20b634716bfbf471eb5cddfd66b

https://github.com/flutter/flutter/assets/1284134/98ce5fab-3718-44ef-bfea-bc71574ca2b3

**Comment by jonahwilliams** [comment]
That's great! So in the future, when you are filling a CP request you should either include more context or file your own issue, because everyone that had previously filled an issue for this is reporting no difference in performance - and we're not going to chery pick changes "just in case", as they are risky in general.

I'm not going to re-open this particular bug (because of complexity of tracking closed pull requests), but please file a new issue and attach more details that would allow us to evaluate the impact of this change.

**Comment by stx** [comment]
The problem, impact, videos, commits, issues and PRs are provided in this issue. Can you explain what context is missing?

Can you explain why filing our own issue, as we have done here, is not considered filing our own issue?

**Comment by jonahwilliams** [comment]
So one of https://github.com/flutter/flutter/issues/131567 or https://github.com/flutter/flutter/issues/132735 are your bugs?

**Comment by stx** [comment]
> So one of #131567 or #132735 are your bugs?

If you're asking if these are the bugs we're affected by, then yes. If you're asking if we were the original creator of the bugs, then no. As you commented [here](https://github.com/flutter/flutter/issues/132735#issuecomment-1802272724), your PR is a 4X improvement on the latter which lines up pretty well with the videos.

**Comment by stx** [comment]
> IOW, if this change is important enough for you, starting with an issue that contains some context like [#138865 (comment)](https://github.com/flutter/flutter/issues/138865#issuecomment-1850942169) is going to be helpful in order to get it approved. Otherwise its easy to write off the change as not impactful enough to be worth the risk.
> 
> In the template you've filled out the following line:
> 
> > Impact Description
> > Bad jank compared to Skia when multiple blurs are present on screen.
> 
> Linking to other folks bugs, bugs in which they claim this was **not** a substantial improvement for their own application, is not sufficent for this. nor is linking to improvements in flutter's own artificial benchmarks. Almost anyone looking at this cherry pick request and evaluating it is going to decline it.

Okay. So to get ours issues addressed, we need to duplicate the existing issues from other users, describe our own personal impact with them, and then reference those new issues in new CP requests? Is this correct?

**Comment by jonahwilliams** [comment]
That's not what I said. I think my time attempting to help here has run its course. If you disagree feel free to discuss on the discord or file a new issue.
